benefits/core/models/enrollment.py

```
@@ -257,8 +258,12 @@ def enrollment_success_template(self):
         else:
             return self.enrollment_success_template_override or f"{prefix}--{self.transit_agency.slug}.html"
 
+    @property
+    def in_person_eligibility_context(self):
+        return in_person_context.eligibility_index[self.system_name].dict()
```

@thekaveman I rewrote this so that it is non-defensive and simply looks up the key. Consumers of this property are responsible for knowing what to do, whether that's to let the exception bubble up or to handle it.

Here's the diff: https://github.com/cal-itp/benefits/pull/2689/commits/21e0f69d52d7331e5cce87d1be15b2b2e9fce5a2

That commit shows how `InPersonEligibilityForm` consumes the property. [`EnrollmentFlow.clean`](https://github.com/cal-itp/benefits/pull/2689/commits/d27a53e995573b6213412f8f63711b5d102c3ebf#diff-396f633e76d6f72ae9c1c006143cd627138a10b7773649831c0db29c8dabf627R286-R290) consumes it in a similar way.

---

Ah, I see... yeah I don't think this is the specific use-case I was thinking about, and I didn't understand the difference in what we were talking about. I agree this change isn't as clean.

The use-case I was thinking about (related to https://github.com/cal-itp/benefits/pull/2698#issuecomment-2672430934): Why was it possible to load a fixture with a bad `system_name`? I expected this to have failed on import: https://github.com/cal-itp/benefits/pull/2698#issuecomment-2672297744

But I think you described why this is: https://github.com/cal-itp/benefits/pull/2698#issuecomment-2672384526. And then it was fixed anyway by adding `agency_card` to the enum...

Sorry, I mixed up a few different things in that comment. But really I was wondering: why should we even be able to import bad data like that? And does that mean it is somehow possible to get a bad value in there from the Admin? I think both of those are probably answered by your explanation.

---

Just to resolve this, the way this is implemented is just fine if you don't want to worry about it anymore :+1:

---

Thanks, yeah I think this is fine too